### PR TITLE
Add metrics and improve telemetry support in Azure.AI.OpenAI

### LINKED ISSUE #5451: OpenAI component doesn't setup telemetry correctly
I have been fiddling with the playground sample for the playground OpenAI component trying to get it to emit telemetry. I think the following things are missing;

- Updating Directory.Packages.props to use beta3 for the Azure.AI.OpenAI package

``` xml
<PackageVersion Include="Azure.AI.OpenAI" Version="2.0.0-beta.3" />
```

I also had to add a source to nuget as that version is pulling in a more recent OpenAI SDK version than is available internally at the moment

- In the Aspire.Azure.AI.OpenAI component:
  -  Update the activity source names to include OpenAI.*

``` csharp
protected virtual string[] ActivitySourceNames => new[] { $"{typeof(TClient).Namespace}.*", "OpenAI.*" }; 
```

  - Set the app context switch for the OpenAI instrumentation

``` csharp
        if (GetTracingEnabled(settings))
        {
            AppContext.SetSwitch("OpenAI.Experimental.EnableOpenTelemetry", true);
            builder.Services.AddOpenTelemetry()
                .WithTracing(traceBuilder => traceBuilder.AddSource(ActivitySourceNames));

        }
```

When these are all in place, we get the metrics and telemetry that @lmolkova added to OpenAI recently. 

image

I believe we should include the app context switch - its there because the semanic conventions are not stable, but IMHO for Aspire we should be showing what's available rather than hiding it.

### PR DESCRIPTION
**Pull request by sebastienros**
## Description

- Enables telemetry app switch when metrics or tracing is configured.
- Adds support for metrics in Azure components.
- Enables metrics in Azure.AI.OpenAI.

Covers #5451

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
  - [ ] No
- Did you add public API?
  - [x] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [x] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [x] Yes
      - [ ] No
  - [ ] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Link to aspire-docs issue: 
  - [x] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5999)

### CODE REVIEW

File: src/Components/Aspire.OpenAI/AspireOpenAIExtensions.cs
Comment (eerhardt) [inline src/Components/Aspire.OpenAI/AspireOpenAIExtensions.cs:110]: I think we should be following the same pattern as the other experimental telemetry in the Azure SDK - 

https://github.com/dotnet/aspire/blob/25bbf0543a5915d2ff4b5055431bce176bf1c650/src/Components/Aspire.Azure.Messaging.ServiceBus/AzureMessagingServiceBusSettings.cs#L64-L80

cc @lmolkova

Comment (sebastienros) [reply src/Components/Aspire.OpenAI/AspireOpenAIExtensions.cs]: Applied this pattern here and also did that for OpenAI from which I copied it initially.
Note that it doesn't make it automatic then and users still have to nabe the flag even though we advertise telemetry is on by default. But I assume it's on purpose because it's experimental. Not sure the issue that is linked should be marked as fixed then after that.

Comment (eerhardt) [reply src/Components/Aspire.OpenAI/AspireOpenAIExtensions.cs]: > and users still have to nabe the flag even though we advertise telemetry is on by default. But I assume it's on purpose because it's experimental.

Correct. It is experimental, so we should let the users decide when they want to enable it (and become dependent on an experimental feature).

File: tests/Aspire.Azure.AI.OpenAI.Tests/ConformanceTests.cs
Comment (eerhardt) [inline tests/Aspire.Azure.AI.OpenAI.Tests/ConformanceTests.cs:107]: (nit) the whitespacing/indent here could be cleaned up.

File: src/Components/Aspire.Azure.AI.OpenAI/AzureOpenAISettings.cs
Comment (eerhardt) [inline src/Components/Aspire.Azure.AI.OpenAI/AzureOpenAISettings.cs:54]: It's not necessarily `false` - it defaults to the value of the env var / appcontext switch. I think we could just leave this `value` section off until the env var goes away.

**Review by eerhardt — APPROVED** [review]
LGTM. Thanks!

File: src/Components/Aspire.Azure.AI.OpenAI/AzureOpenAISettings.cs
Comment (sebastienros) [reply src/Components/Aspire.Azure.AI.OpenAI/AzureOpenAISettings.cs:54]: applied

**Comment by sebastienros** [comment]
Oops, I think I should not check auto-merge unless it's only waiting for a build to pass, and not for also an approval. Following up with a PR to fix the doc.

**Pull request merged**
